examples/bevy_utilities/src/noise.rs

```diff
@@ -44,32 +80,7 @@ pub fn generate_noise_chunks3(
         for p in chunks_extent.0.iter_points() {
             s.spawn(async move {
                 let chunk_min = p * chunk_shape;
-                let chunk_extent = Extent3i::from_min_and_shape(chunk_min, chunk_shape);
-
-                let mut array = noise_array3(chunk_extent, freq, seed, octaves);
-
-                if subsurface_only {
-                    let mut any_negative = false;
-                    array.for_each_mut(&chunk_extent, |_: (), x| {
-                        *x *= scale;
-
-                        if *x < 0.0 {
-                            any_negative = true;
-                        }
-                    });
-
-                    if any_negative {
-                        Some((chunk_min, array))
-                    } else {
-                        None
-                    }
-                } else {
-                    array.for_each_mut(&chunk_extent, |_: (), x| {
-                        *x *= scale;
-                    });
-
-                    Some((chunk_min, array))
-                }
+                generate_noise_chunk3(chunk_min, chunk_shape, freq, scale, seed, octaves, subsurface_only).and_then(|chunk| Some((chunk_min, chunk)))
```

Comment:
I think `Option::map` is better than `and_then` for this since you are always returning `Some`.

Reply:
Done.